README.md

```diff
@@ -161,6 +130,51 @@ default values.
   receive globally unique search results only once, then use this
   configuration variable, or use `setopt HIST_IGNORE_ALL_DUPS`.
 
+The following variables must be overridden before having loaded this script
+into your ZSH session, and they define the `history-substring-search-up` and
+`history-substring-search-down` key bindings.
+
+* `HISTORY_SUBSTRING_SEARCH_MAIN_UP_KEYS` is a global array that defines the
+  main keymap keys to be bind to the `history-substring-search-up` command.
+  Its default value is `('^[[A' $terminfo[kcuu1])`, which are two common codes
+  for the UP key in most terminals.
+
+* `HISTORY_SUBSTRING_SEARCH_MAIN_DOWN_KEYS` is a global array that defines the
+  main keymap keys to be bind to the `history-substring-search-down` command.
+  Its default value is `('^[[B' $terminfo[kcud1])`, which are two common codes
+  for the DOWN key in most terminals.
+
+* `HISTORY_SUBSTRING_SEARCH_EMACS_UP_KEYS` is a global array that defines the
+  EMACS mode keys to be bind to the `history-substring-search-up` command.
+  Its default value is `('^P')`, which is the code for the Control-P key.
+
+* `HISTORY_SUBSTRING_SEARCH_EMACS_DOWN_KEYS` is a global array that defines the
+  EMACS mode keys to be bind to the `history-substring-search-down` command.
+  Its default value is `('^N')`, which is the code for the Control-N key.
+
+* `HISTORY_SUBSTRING_SEARCH_VICMD_UP_KEYS` is a global array that defines the
+  VI mode keys to be bind to the `history-substring-search-up` command.
+  Its default value is `('k')`, which is the code for the `k` key.
+
+* `HISTORY_SUBSTRING_SEARCH_VICMD_DOWN_KEYS` is a global array that defines the
+  VI mode keys to be bind to the `history-substring-search-down` command.
+  Its default value is `('j')`, which is the code for the `j` key.
```

Comment:
I think there's lots of repeating words in the above paragraphs, but I tried to keep the bullets layout. Maybe this could be reformatted as a table?

Reply:
True.  I think the repetition is fine.  If you really want to reformat this as a table, please make sure to do the same for all other configuration variables too.

Reply:
Let's keep the repetition here then, as you're fine with it. I can try a table layout later in a separate PR...